runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcDispatcherClient.java

```diff
@@ -45,92 +48,156 @@ class GrpcDispatcherClient {
   private final WindmillStubFactory windmillStubFactory;
 
   @GuardedBy("this")
-  private final List<CloudWindmillServiceV1Alpha1Stub> dispatcherStubs;
-
-  @GuardedBy("this")
-  private final Set<HostAndPort> dispatcherEndpoints;
+  private final AtomicReference<DispatcherStubs> dispatcherStubs;
 
   @GuardedBy("this")
   private final Random rand;
 
   private GrpcDispatcherClient(
       WindmillStubFactory windmillStubFactory,
-      List<CloudWindmillServiceV1Alpha1Stub> dispatcherStubs,
-      Set<HostAndPort> dispatcherEndpoints,
+      AtomicReference<DispatcherStubs> dispatcherStubs,
```

Comment:
just take in DispatcherStubs and wrap in atomic reference?
Seems to otherwise expose unencessarily how the stubs is used.